pkg/in/tcp.go

```diff
@@ -75,20 +76,29 @@ func readFromConnectionTCPBuf(wg *sync.WaitGroup, conn net.Conn, queue chan<- []
 			zap.String("sender", conn.RemoteAddr().String()))
 	}
 
-	scanForRecordsTCPBuf(conn, queue, cfg, ms, lg)
+	scanForRecordsTCPBuf(conn, queue, rateLimiters, cfg, ms, lg)
 }
 
-func scanForRecordsTCPBuf(conn net.Conn, queue chan<- [][]byte, cfg *conf.Main, ms *metrics.Prom, lg *zap.Logger) {
+func scanForRecordsTCPBuf(conn net.Conn, queue chan<- [][]byte, rateLimiters []ratelimiter.RateLimiter, cfg *conf.Main, ms *metrics.Prom, lg *zap.Logger) {
 	sc := bufio.NewScanner(conn)
 
 	buf := make([]byte, 2048)
 	sc.Buffer(buf, bufio.MaxScanTokenSize)
-
 	qb := NewBatchChan(queue, int(cfg.MainQueueBatchSize), int(cfg.BatchFlushPerdiodSec), ms)
 	defer qb.Close()
 
+	var rlTickerChan <-chan time.Time
+	if rateLimiters != nil && cfg.RateLimiterIntervalMs > 0 {
```

Comment:
It looks like the setting `RateLimiterIntervalMs` plays a double role. Its second role is to enable the limiter, but this role is already covered by the `RateLimiterEnabled` option. So, we have a duplication of the limiter on/off switch. We either need to remove the `RateLimiterEnabled` option and use this one as a switch, or we need to add a config validity check and consider a config with `RateLimiterIntervalMs` set to 0 as invalid. Please let me know what you think.